app/src/main/java/com/orange/ods/app/ui/components/ComponentsNavGraph.kt

```diff
@@ -20,38 +25,69 @@ import com.orange.ods.app.ui.MainDestinations
 import com.orange.ods.app.ui.MainTopAppBarState
 
 fun NavGraphBuilder.addComponentsGraph(navigateToElement: (String, Long?, NavBackStackEntry) -> Unit) {
+
     composable(
         "${MainDestinations.ComponentDetailRoute}/{${MainDestinations.ComponentIdKey}}",
         arguments = listOf(navArgument(MainDestinations.ComponentIdKey) { type = NavType.LongType })
     ) { from ->
-        LocalMainTopAppBarManager.current.reset()
-
         val arguments = requireNotNull(from.arguments)
-        val componentId = arguments.getLong(MainDestinations.ComponentIdKey)
-        ComponentDetailScreen(
-            componentId = componentId,
-            onVariantClick = { variantId -> navigateToElement(MainDestinations.ComponentVariantRoute, variantId, from) },
-            onDemoClick = { navigateToElement(MainDestinations.ComponentDemoRoute, componentId, from) }
-        )
+        var currentComponentId: Long by remember { mutableStateOf(-1) }
+        val routeComponentId = arguments.getLong(MainDestinations.ComponentIdKey)
+        val hasComponentIdChanged by remember {
+            derivedStateOf { currentComponentId != routeComponentId }
+        }
+
+        val component = remember(routeComponentId) { components.firstOrNull { component -> component.id == routeComponentId } }
+        component?.let {
+            if (hasComponentIdChanged) {
```

Comment:
I tried removing `hasComponentIdChanged` and it only added one recomposition of the MainTopAppBar. If this is correct and gain is only one recomposition, I think we can simplify the code and remove all the stuff related to `hasComponentIdChanged`.

Same comment for `hasVariantIdChanged`.